### PR TITLE
Fixes several problems found during authz testing [skip ci]

### PR DESCRIPTION
**Pull request by nvidianz**
1. Fixed clone_job, not copying all fields.
2. Fixed the deploy error when one site rejects the job.
3. Admin API errors are logged now, instead of causing FOBS error
4. Fixed Workspace decomposer problem caused by name->site_name change.

### CODE REVIEW

File: nvflare/private/fed/server/server_engine.py [hunk 1]
```diff
@@ -726,11 +726,16 @@ def check_client_resources(self, resource_reqs) -> Dict[str, Tuple[bool, str]]:
         for r in replies:
             site_name = self.get_client_name_from_token(r.client_token)
             if r.reply:
-                resp = fobs.loads(r.reply.body)
-                result[site_name] = (
-                    resp.get_header(ShareableHeader.CHECK_RESOURCE_RESULT, False),
-                    resp.get_header(ShareableHeader.RESOURCE_RESERVE_TOKEN, ""),
-                )
+                error_code = r.reply.get_header(MsgHeader.RETURN_CODE, ReturnCode.OK)
```
Comment (yanchengnv) [inline nvflare/private/fed/server/server_engine.py:729]: This is good fix. There is a similar case (Line 584) in aux_send(). Please fix it.

**Review by yanchengnv — CHANGES_REQUESTED** [review]
Looks really good. Just need one more fix for aux_send.

File: nvflare/private/fed/server/job_cmds.py [hunk 2]
```diff
@@ -344,12 +355,15 @@ def clone_job(self, conn: Connection, args: List[str]):
             with engine.new_context() as fl_ctx:
                 data_bytes = job_def_manager.get_content(job_id, fl_ctx)
 
+                job_meta = {str(k): job.meta[k] for k in job.meta.keys() & CLONED_META_KEYS}
+
```
[the remark this answers is not in the document]
Comment (nvidianz) [reply nvflare/private/fed/server/job_cmds.py:359]: Yep. & is to get the intersection of 2 sets. When cloning jobs, we don't want to copy things like deploy_details.

File: nvflare/private/fed/server/job_runner.py [hunk 3]
```diff
@@ -338,10 +338,16 @@ def run(self, fl_ctx: FLContext):
...
+                                else:
+                                    deployable_clients = client_sites
+
                                 self._start_run(
                                     job_id=job_id,
                                     job=ready_job,
-                                    client_sites=client_sites,
+                                    client_sites=deployable_clients,
                                     fl_ctx=fl_ctx,
                                 )
```
Comment (YuanTingHsieh) [inline nvflare/private/fed/server/job_runner.py:352]: Maybe we need to check if a job is deployable at scheduler time.

Cause we need to also check if require_clients is in deployable_clients
and if deployable_clients are larger than min_clients

Comment (YuanTingHsieh) [reply nvflare/private/fed/server/job_runner.py:352]: got it, make sense to me, agree to keep this for now.

File: nvflare/private/fed/server/server_engine.py (hunk 1 above)
Comment (nvidianz) [reply nvflare/private/fed/server/server_engine.py:733]: That error is also serialized so it doesn't actually cause a FOBS error. I added logging when error reply is received.

**Review by YuanTingHsieh — APPROVED** [review]
thanks, LGTM

**Review by yhwen — DISMISSED** [review]

File: nvflare/private/fed/server/server_engine.py [hunk 4]
```diff
@@ -581,12 +581,16 @@ def aux_send(self, targets: [], topic: str, request: Shareable, timeout: float,
             client_name = self.get_client_name_from_token(r.client_token)
             if r.reply:
                 try:
-                    results[client_name] = fobs.loads(r.reply.body)
-                except BaseException:
+                    shareable = fobs.loads(r.reply.body)
```
Comment (yanchengnv) [inline nvflare/private/fed/server/server_engine.py:584]: We should do exactly the same thing as line 733: check return_code first. Only if the RC is OK, should we do fobs.loads().

**Review by yanchengnv — CHANGES_REQUESTED** [review]

File: nvflare/private/fed/server/server_engine.py (hunk 4 above)
Comment (nvidianz) [reply nvflare/private/fed/server/server_engine.py]: Done.

**Review by yanchengnv — APPROVED** [review]

**Pull request merged**